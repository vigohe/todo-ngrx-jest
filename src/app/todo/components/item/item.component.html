<h1 [style.text-decoration]="todo.complete ? 'line-through' : ''">
    {{todo.text}} <input type="checkbox" (change)="onChecked(todo)" [checked]="todo.complete"/> <button (click)="onRemove(todo)">X</button>
</h1>
