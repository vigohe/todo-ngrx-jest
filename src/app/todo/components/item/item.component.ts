import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { Todo } from '../../models/todo';

@Component({
  selector: 'app-item',
  templateUrl: './item.component.html',
  styleUrls: ['./item.component.css']
})
export class ItemComponent implements OnInit {

  @Input() todo: Todo;
  @Output() toggle = new EventEmitter<Todo>();
  @Output() remove = new EventEmitter<Todo>();

  constructor() { }

  ngOnInit() {
  }

  onChecked(todo: Todo) {
    this.toggle.emit(todo);
  }

  onRemove(todo: Todo) {
    this.remove.emit(todo);
  }

}
