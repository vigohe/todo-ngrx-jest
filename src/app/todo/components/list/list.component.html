<span *ngFor="let todo of todos; let i=index">
  <app-item [todo]="todo" (toggle)="onToggle($event)" (remove)="onRemove($event)"></app-item>
</span>
