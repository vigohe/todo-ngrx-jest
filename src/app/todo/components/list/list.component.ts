import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { Todo } from '../../models/todo';

@Component({
  selector: 'app-list',
  templateUrl: './list.component.html',
  styleUrls: ['./list.component.css']
})
export class ListComponent implements OnInit {
  @Input() todos: Todo[];
  @Output() toggle = new EventEmitter<Todo>();
  @Output() remove = new EventEmitter<Todo>();

  constructor() { }

  ngOnInit() {
  }

  onToggle(todo: Todo) {
    this.toggle.emit(todo);
  }

  onRemove(todo: Todo) {
    this.remove.emit(todo);
  }

}
