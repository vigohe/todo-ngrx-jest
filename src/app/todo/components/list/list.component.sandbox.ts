import { sandboxOf } from 'angular-playground';
import { ListComponent } from './list.component';
import { Todo } from '../../models/todo';
import { ItemComponent } from '../item/item.component';

export default sandboxOf(ListComponent,{declarations: [ItemComponent]})
  .add('with simple text', {
    template: `<app-list [todos]="todos"></app-list>`,
    context: {
      todos: [
        {'id': 1, 'text': 'GO TO WORK', 'complete': true},
        {'id': 2, 'text': 'LAUNCH', 'complete': false}]
    }
  });
