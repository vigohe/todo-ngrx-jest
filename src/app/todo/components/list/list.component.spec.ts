import {ListComponent} from './list.component';
import {ItemComponent} from '../item/item.component';
import { ComponentFixture, TestBed } from '@angular/core/testing';


describe('ListComponent', () => {
  let component: ListComponent;
  let fixture: ComponentFixture<ListComponent>;

  beforeEach(async() => {
    TestBed.configureTestingModule({
      imports: [],
      declarations: [ ListComponent, ItemComponent]
    });

    await TestBed.compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(ListComponent);
    component = fixture.componentInstance;

    fixture.detectChanges();
  });


  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should compile', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
