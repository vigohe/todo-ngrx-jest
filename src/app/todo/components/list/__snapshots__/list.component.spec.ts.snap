// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListComponent should compile 1`] = `
<app-list
  remove={[Function EventEmitter]}
  toggle={[Function EventEmitter]}
>
  
  
</app-list>
`;
